### PR TITLE
Add support and tests for SSL on custom domains in CDN

### PR DESCRIPTION
**Pull request by nachakra**
Link to swagger PR - https://github.com/Azure/azure-rest-api-specs/pull/5276

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
You are missing Swagger PR information.
Paste a link to the swagger PR.
Also update nuget package version in .csproj, also update version in AssemblyInfo.cs
Generate SDK using generate.ps1, which will essentially generate all the meta data files that are required for .NET SDK

**Comment by nachakra** [comment]
> You are missing Swagger PR information.
> Paste a link to the swagger PR.
> Also update nuget package version in .csproj, also update version in AssemblyInfo.cs
> Generate SDK using generate.ps1, which will essentially generate all the meta data files that are required for .NET SDK

**Comment by nachakra** [comment]
@shahabhijeet -
1) I added the swagger link to the description. 
2) I updated the assemblyinfo.cs. When do we need to change the assemblyinfo.cs if it is not a breaking change. Also it is not a new version of API. 
.csproj is included in the PR and the nuget package version is already updated.
3) I did use the generate.cmd to generate the SDK and I do see a metadata file in this PR - /SDKs/_metadata/cdn_resource-manager.txt
Should there be more?

File: src/SDKs/Cdn/Cdn.Tests/ScenarioTests/CustomDomainTests.cs [hunk 1]
```diff
@@ -112,17 +112,40 @@ public void CustomDomainCRUDTest()
...
 
-                // Delete first custom domain on stopped endpoint should succeed
+                // add the same deleted custom domain again
+                cdnMgmtClient.CustomDomains.Create(resourceGroupName, profileName, endpointName, customDomainName2, "sdk-2-f3757d2a3e10.DUSTYDOGPETCARE.US");
+
+                // Enable custom https using BYOC on custom domain 
+                var byocParameters = new KeyVaultCertificateSourceParameters()
+                {
+                    ResourceGroupName = "byoc",
+                    SecretName = "CdnSDKE2EBYOCTest",
```
Comment (shahabhijeet) [inline src/SDKs/Cdn/Cdn.Tests/ScenarioTests/CustomDomainTests.cs]: @nachakra no hardcoding secrets.
Please use TestFramework to inject custom data.
https://github.com/Azure/azure-sdk-for-net/blob/master/Documentation/Using-Azure-TestFramework.md#1-once-you-set-your-connection-string-you-can-add-or-update-keyvalue-settings

File: src/SDKs/Cdn/Management.Cdn/Generated/CdnManagementClient.cs [hunk 2]
```diff
@@ -397,6 +397,8 @@ private void Initialize()
             DeserializationSettings.Converters.Add(new PolymorphicDeserializeJsonConverter<DeliveryRuleAction>("name"));
             SerializationSettings.Converters.Add(new PolymorphicSerializeJsonConverter<DeliveryRuleCondition>("name"));
             DeserializationSettings.Converters.Add(new PolymorphicDeserializeJsonConverter<DeliveryRuleCondition>("name"));
+            SerializationSettings.Converters.Add(new PolymorphicSerializeJsonConverter<CustomDomainHttpsParameters>("certificateSource"));
```
Comment (shahabhijeet) [inline src/SDKs/Cdn/Management.Cdn/Generated/CdnManagementClient.cs]: @nachakra did you add this after the generated code, if yes, all these custom settings should be added inside CustomInitialize() that you will be defining in your customization layer.
Create a customization folder and add a new class file with the same exact class name (as all classes partial)
Declare CustomInitialize() in it and add whatever custom code you need.

File: src/SDKs/Cdn/Management.Cdn/Microsoft.Azure.Management.Cdn.csproj [hunk 3]
```diff
@@ -5,11 +5,11 @@
 
 	<PropertyGroup>
 		<Description>Microsoft Azure CDN Management Library</Description>
-		<Version>4.0.2-preview</Version>
+		<Version>4.0.3-preview</Version>
```
Comment (shahabhijeet) [inline src/SDKs/Cdn/Management.Cdn/Microsoft.Azure.Management.Cdn.csproj]: you added a new functionality.
Update minor version, so make
nuget version = 4.1.2-preview
assemblyfileversion = 4.1.2

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Cdn/Cdn.Tests/ScenarioTests/CustomDomainTests.cs (hunk 1 above)
Comment (nachakra) [reply src/SDKs/Cdn/Cdn.Tests/ScenarioTests/CustomDomainTests.cs]: This is the cert name in Keyvault and not a secret :). That's how the property is called in the API model.

File: src/SDKs/Cdn/Management.Cdn/Generated/CdnManagementClient.cs (hunk 2 above)
Comment (nachakra) [reply src/SDKs/Cdn/Management.Cdn/Generated/CdnManagementClient.cs]: I did not change anything in the generated code. Just refactored the tests.

File: src/SDKs/Cdn/Management.Cdn/Microsoft.Azure.Management.Cdn.csproj (hunk 3 above)
Comment (nachakra) [reply src/SDKs/Cdn/Management.Cdn/Microsoft.Azure.Management.Cdn.csproj]: Sure.

**Comment by nachakra** [comment]
@shahabhijeet - can you take a look again at the PR please?

**Review by dsgouda — APPROVED** [review]
LGTM

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
If you are planning to release this SDK, please update version info in .csproj and AssemblyInfo.cs
Also add release notes.

**Comment by nachakra** [comment]
@shahabhijeet  - I updated both in the last commit or the commits before. Not sure what I am still missing.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**